src/snowflake/snowpark/session.py

```diff
@@ -191,6 +191,16 @@ class Session:
     A :class:`Session` object is not thread-safe.
     """
 
+    class RuntimeConfig:
+        def __init__(self) -> None:
+            self._conf: Dict[str:Any] = {}
+
+        def get(self, key: str, default=None):
+            return self._conf.get(key, default)
+
+        def unset(self, key: str):
```

Comment:
Is it correct that we don't support `set` because we want to make it immutable?

Reply:
If we update `conf`, even using `unset`, should we also change https://github.com/snowflakedb/snowpark-python/blob/30530f8c644c40634d42cf9b95f7895dcbb4647f/src/snowflake/snowpark/_internal/server_connection.py#L136? I think we should make them the same any time